### PR TITLE
Prettier

### PR DESCRIPTION
**Pull request by menor**
Add prettier and precommit and prepush tasks
Most of the stuff are automatic changes prettier did.
Now we lint and reformat the code before commiting
And run the tests before pushing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,12 +10,16 @@
     "build:babel": "NODE_ENV=release babel ./src --out-dir ./dist --ignore stories.js,spec.js,tape.js",
     "build:icon-index": "babel-node ./scripts/icon-index-generator.js",
     "clean:build": "rimraf dist",
+    "precommit": "yarn run prettier && yarn run lint",
```
Comment (heinburger) [inline package.json]: 👍

File: src/components/Dropdown/Dropdown.js [hunk 2]
```diff
@@ -115,37 +130,39 @@ class Dropdown extends Component {
...
-    const {openDropdown, closeDropdown, isOpen} = this.props
-    return (openDropdown !== undefined && closeDropdown !== undefined && isOpen !== undefined)
+    const { openDropdown, closeDropdown, isOpen } = this.props
+    return (
+      openDropdown !== undefined &&
+      closeDropdown !== undefined &&
+      isOpen !== undefined
+    )
   }
 }
```
Comment (heinburger) [inline src/components/Dropdown/Dropdown.js]: this looks like an older version of some of these components

Comment (menor) [inline src/components/Dropdown/Dropdown.js]: Oh, maybe I branched out before it was on master, no problem I can checkout just the dropdown folder from there.

Comment (heinburger) [inline src/components/Dropdown/Dropdown.js]: ah ok - this is going into v2-beta - didn't see that :) - I will merge now

**Pull request merged**